src/page/blog/review-page.jsx

```diff
@@ -35,15 +35,25 @@ export const Review = ({ id }) => {
 
   useEffect(() => {
     if (data) {
-      setPost({
-        title: data.post.title,
-        content: data.post.content,
-        username: data.post.author.username,
-        keywords: data.post.keywords,
-        createdAt: data.post.createdAt,
-      });
-      if (currentUser.id === data.post.author.id) {
-        setAccessGranted(true);
+      if (data.post.author) {
```

Comment:
Another way to go about this is get rid of the `if`, and use optional chaining to not break the JS (ie `data.post?.author?.username`). This would lead to less code overall. Then adjust the conditional below to account for undefined. But overall this is working great!